test/Table_Tests/src/Table_Spec.enso

```diff
@@ -617,4 +619,31 @@ spec =
             problems = [Duplicate_Output_Column_Names_Data ["A", "A", "A"]]
             Problems.test_problem_handling action problems tester
 
+    Test.group "[In-Memory] Table.aggregate" <|
+        Test.specify "should return columns with correct types" <|
+            dates = ["dates", [Date.new 1999, Date.new 2000, Date.new 2000, Date.new 2000]]
+            texts = ["texts", ["a", "bb", "a", "bb"]]
+            mixed = ["mixed", [1, "a", "a", 1]]
+            ints = ["ints", [0, 1, 1, 0]]
+            floats = ["floats", [0.1, 1.0, 2.0, 1.5]]
+            objects = ["objects", [My_Data 0 1, My_Data 0 1, My_Data 2 2, My_Data 2 2]]
+            table = Table.new [dates, texts, mixed, ints, floats, objects]
+
+            t1 = table.aggregate [Group_By "dates", Shortest "texts", Aggregate_Column.First "texts", Aggregate_Column.First "objects", Aggregate_Column.First "ints", Aggregate_Column.Last "mixed"]
+            t1.info.at "Column" . to_vector . should_equal ["dates", "Shortest texts", "First texts", "First objects", "First ints", "Last mixed"]
+            t1.info.at "Storage Type" . to_vector . should_equal [Storage.Date, Storage.Text, Storage.Text, Storage.Any, Storage.Integer, Storage.Any]
+
+            t2 = table.aggregate [Mode "dates", Count_Distinct "objects", Count_Distinct "texts", Minimum "ints", Maximum "floats"]
+            t2.info.at "Column" . to_vector . should_equal ["Mode dates", "Count Distinct objects", "Count Distinct texts", "Minimum ints", "Maximum floats"]
+            t2.info.at "Storage Type" . to_vector . should_equal [Storage.Date, Storage.Integer, Storage.Integer, Storage.Integer, Storage.Decimal]
+
+            t3 = table.aggregate [Group_By "texts", Group_By "ints", Aggregate_Column.Last "floats"]
+            t3.info.at "Column" . to_vector . should_equal ["texts", "ints", "Last floats"]
+            t3.info.at "Storage Type" . to_vector . should_equal [Storage.Text, Storage.Integer, Storage.Decimal]
+
+            t4 = table.aggregate [Group_By "mixed", Sum "ints", Sum "floats"]
+            t4.info.at "Column" . to_vector . should_equal ["mixed", "Sum ints", "Sum floats"]
+            t4.info.at "Storage Type" . to_vector . should_equal [Storage.Any, Storage.Decimal, Storage.Decimal]
```

Comment:
```suggestion
            t4.info.at "Storage Type" . to_vector . should_equal [Storage.Any, Storage.Integer, Storage.Decimal]
```

I'm thinking that it may make sense for sum of integers to still be integer - but that can have its overflow problems. So I didn't want to get sidetracked for too long into this and kept it as is. I think it's probably something to revisit once we get the more precise types.